### PR TITLE
221.Add force checkout and compile functionality 

### PR DESCRIPTION
**Pull request by singhd789**
## Checklist before requesting a review

- [x] I ran my code
- [x] I tried to make my code readable
- [x] I tried to comment my code
- [ ] I wrote a new test, if applicable
- [ ] I wrote new instructions/documentation, if applicable
- [ ] I ran pytest and inspected it's output
- [ ] I ran pylint and attempted to implement some of it's feedback

### CODE REVIEW

**Comment by ilaflott** [comment]
Looks fairly well thought out to me here- any tests coming? 

maybe add quick desc of what the PR adds in plain english when you can!

**Comment by singhd789** [comment]
> Looks fairly well thought out to me here- any tests coming?
> 
> maybe add quick desc of what the PR adds in plain english when you can!

Oh wait! I can totally add test to test_run_fremake since that's in there already. Also good idea on the description

**Comment by singhd789** [comment]
@ilaflott Thanks for taking a look in your spare time btw

**Comment by ilaflott** [comment]
i'd help resolve these conflicts, approve+merge whenever you're ready.

**Comment by singhd789** [comment]
> i'd help resolve these conflicts, approve+merge whenever you're ready.

Thanks @ilaflott. I'll work on merge conflicts and probably ping you when I think everything's good

**Comment by singhd789** [comment]
@ilaflott @uramirez8707 @rem1776  Ooook, had to iron out some things that I was doing that I actually didn't want to be doing, but I think this PR should be ok for reviews/tests now :crossed_fingers:

**Review by uramirez8707 — COMMENTED** [review]
I tried these options and they seem to work. 

Just have some comments.

File: fre/make/create_checkout_script.py [hunk 1]
```diff
@@ -37,10 +65,16 @@ def checkout_create(yamlfile, platform, target, no_parallel_checkout, jobs, exec
...
+    # If force-checkout defined: re-combine model, compile, and platform yamls
+    if force_checkout:
+        print("Re-combine yaml files")
+        comb = cy.init_compile_yaml(yml,platform,target)
+        full_combined = cy.get_combined_compileyaml(comb)
+    else:
+        ## If combined yaml exists, note message of its existence
+        ## If combined yaml does not exist, combine model, compile, and platform yamls
+        combined = Path(f"combined-{name}.yaml")
+        full_combined = cy.combined_compile_existcheck(combined,yml,platform,target)
```
[the remark this answers is not in the document]
Comment (rem1776) [reply fre/make/create_checkout_script.py:77]: I agree with this 200%.

This is making me wonder though, why not go one step further and just not even write the combined yaml to a file?

I will admit, I never really understood why we combine the yamls and write it out in the first place, so maybe there's a better reason then I'm seeing.

But if the current fremake yamls (platform/compile/etc) are what we are going to be modifying and using in place of xmls, the combined yamls don't really need to exist as files, they can stay in memory. Otherwise, they just kinda get in the way of whatever yaml files the user specifies.

File: fre/make/create_checkout_script.py [hunk 2]
```diff
@@ -1,21 +1,49 @@
...
+    """
+    fre_checkout = checkout.checkout("checkout.sh",src_dir)
+    fre_checkout.writeCheckout(model_yaml.compile.getCompileYaml(),jobs,pc)
+    fre_checkout.finish(model_yaml.compile.getCompileYaml(),pc)
+
+    # Make checkout script executable
+    os.chmod(src_dir+"/checkout.sh", 0o744)
+    print("    Checkout script created in "+ src_dir + "/checkout.sh \n")
+
+    return fre_checkout
```
Comment (rem1776) [inline fre/make/create_checkout_script.py:27]: just curious, is this returned for a reason?

File: fre/make/create_checkout_script.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (singhd789) [reply fre/make/create_checkout_script.py:77]: I agree with both of you, this is great input!

@rem1776 -  At first, I was trying to load the yamls and just have/use the dictionary it's parsed as, and then update/add to dictionaries in order to dump a combined yaml file with all the info needed. BUT, the main issue I was seeing here was from the reusable variables. 

The yaml won't even load (it will give "undefined alias" issue) if there is a key that uses a reusable variable that isn't defined in `fre_properties`, these being `name`, `platform`, and `target`, and those also aren't defined in `fre_properties` because they can change (since they're click options that are passed in). 

I think this is related to your confusion on why we even create these combined yamls. If there's a better way, I'm definitely up for it though.

**Comment by ilaflott** [comment]
24 days away from a year old!
